lib/screens/authentication/recover/recover_account_timer/interactor/usecase/remaining_time_state_mapper.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:hashed/domain-shared/page_state.dart';
+import 'package:hashed/screens/authentication/recover/recover_account_found/interactor/viewmodels/current_remaining_time.dart';
+import 'package:hashed/screens/authentication/recover/recover_account_timer/interactor/viewmodels/recover_account_timer_bloc.dart';
+
+///Seconds in a day
+const int _daySecond = 60 * 60 * 24;
+
+///Seconds in an hour
+const int _hourSecond = 60 * 60;
+
+///Seconds in a minute
+const int _minuteSecond = 60;
+
+class RemainingTimeStateMapper {
+  RecoverAccountTimerState mapResultToState(RecoverAccountTimerState currentState) {
+    int days = 0;
+    int hours = 0;
+    int min = 0;
+    int remainingTimeStamp = currentState.timeRemaining;
+
+    ///Calculate the number of days remaining.
+    if (remainingTimeStamp >= _daySecond) {
+      days = remainingTimeStamp ~/ _daySecond;
+      remainingTimeStamp %= _daySecond;
+    }
+
+    ///Calculate remaining hours.
+    if (remainingTimeStamp >= _hourSecond) {
+      hours = remainingTimeStamp ~/ _hourSecond;
+      remainingTimeStamp %= _hourSecond;
+    } else if (days != 0) {
+      hours = 0;
+    }
+
+    ///Calculate remaining minutes.
+    if (remainingTimeStamp >= _minuteSecond) {
+      min = remainingTimeStamp ~/ _minuteSecond;
+      remainingTimeStamp %= _minuteSecond;
+    } else if (hours != 0) {
+      min = 0;
+    }
+
+    return currentState.copyWith(
+      pageState: PageState.success,
+      currentRemainingTime: CurrentRemainingTime(
+        days: days,
+        hours: hours,
+        min: min,
+        sec: remainingTimeStamp,
+      ),
+    );
+  }
```

Comment:
we already had a remaining time state mapper

For this code, it's ideal to make some unit tests to make sure the outcome is what we want. 

Also we need more unit tests anyway. 

I remember the last time around our counter had some issues around the edges, e.g. when there's 1 second remaining it wouldn't go to 0, or it would go to -1, can't remember exactly what it was, but there were bugs in it. 

So best to unit test the whole process and make sure we get the right results at 

25 hours remaining
24 hours remaining
23 hours remaining
59 minutes remaining
59 seconds remaining
0 seconds remaining
-1 seconds remaining

All these can be unit tests, testing for > 1 day, > 1 hour, > 1 minute, 0 seconds, -1 seconds.